main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strings"
+	"sync"
+)
+
+type project struct{}
+
+type Project interface {
+	ReadInput() (<-chan Command, []string)
+	Execute(cmds <-chan Command, input []string) []string
+	PrintResult(res []string)
+}
+
+var app Project
+
+type Command struct {
+	Name     string
+	Argument string
+}
+
+const (
+	Cat  = "cat"
+	Grep = "grep"
+	Sort = "sort"
+)
+
+func init() {
+	app = NewProject()
+}
+
+func NewProject() Project {
+	return &project{}
+}
+
+func Run(p Project) {
+	cmds, out := p.ReadInput()
+	res := p.Execute(cmds, out)
+	p.PrintResult(res)
+}
+
+func (p *project) PrintResult(res []string) {
+	for _, line := range res {
+		fmt.Println(line)
+	}
+}
+
+func (p *project) ReadInput() (<-chan Command, []string) {
+	cmds := stdIn()
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	var filePath string
+	out := make(chan Command, len(cmds))
+
+	go func(cmds []string, filePath *string) {
+		for _, cmd := range cmds {
+			sepCmd := strings.Fields(cmd)
+			if *filePath == "" && (len(sepCmd) == 1 || len(sepCmd) == 2 && sepCmd[0] == Grep) {
+				log.Fatal("filepath not given")
+			} else if len(sepCmd) == 0 {
+				log.Fatal("input is not correct")
+			}
```

Comment:
Do not use `else if` use early return instead. Also, here yoy can simplify the logic and remove complex if statement